### PR TITLE
[AMORO-3102][doc]: upgrade cdc-ingestion doc from flink cdc 3.x

### LINKED ISSUE #3102: [doc]: upgrade cdc-ingestion doc from flink cdc 3.x
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

We need to upgrade some demo for [Apache Flink CDC 3.X](https://nightlies.apache.org/flink/flink-cdc-docs-stable/) with **cdc-ingestion** doc

_We also recommend that users use Flink CDC version 3.x_

include example:
- Ingest into one table
  - use **flink-connector-mysql-cdc** to native iceberg table
  - use **flink-connector-mysql-cdc** to [Mixed-Iceberg format](https://amoro.apache.org/docs/0.7.0/cdc-ingestion/#mixed-iceberg-format) table

- Ingest Into multiple tables
  - use **flink-connector-mysql-cdc** to native iceberg table
  - use **flink-connector-mysql-cdc** to [Mixed-Iceberg format](https://amoro.apache.org/docs/0.7.0/cdc-ingestion/#mixed-iceberg-format) table

- Next What we do ?
  - Apache Flink CDC 3.x pipeline connector to native iceberg
  - Apache Flink CDC 3.x pipeline connector to [Mixed-Iceberg format](https://amoro.apache.org/docs/0.7.0/cdc-ingestion/#mixed-iceberg-format) table

### How should we improve?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by czy006**
## Why are the changes needed?

Close #3102.

## Brief change log

- add flink cdc doc in flink mode
- cdc-ingestion doc change

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (docs)

### CODE REVIEW

**Comment by czy006** [comment]
cc @zhoujinsong

File: docs/engines/flink/flink-cdc-ingestion.md [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+# Apache CDC Ingestion
+CDC stands for Change Data Capture, which is a broad concept, as long as it can capture the change data, it can be called CDC. [Flink CDC](https://github.com/apache/flink-cdc) is a Log message-based data capture tool, all the inventory and incremental data can be captured. Taking MySQL as an example, it can easily capture Binlog data through Debezium and process the calculations in real time to send them to the data lake. The data lake can then be queried by other engines.
+
+This section will show how to ingest one table or multiple tables into the data lake for both [Iceberg](../iceberg-format/) format and [Mixed-Iceberg](../mixed-iceberg-format/) format.
+## Ingest into one table
+### Iceberg format
+The following example will show how [MySQL CDC](https://nightlies.apache.org/flink/flink-cdc-docs-master/docs/connectors/flink-sources/mysql-cdc/) data is written to an Iceberg table.
```
Comment (huyuanfeng2018) [inline docs/engines/flink/flink-cdc-ingestion.md:33]: Maybe the link should be to the specific version?

File: docs/engines/flink/flink-cdc-ingestion.md [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+  }
+}
+```
+
+### Mixed-Iceberg format
+The following example will show how to write CDC data from multiple MySQL tables into the corresponding Mixed-Iceberg table.
+
+**Requirements**
+
+Please add [Flink Connector MySQL CDC](https://mvnrepository.com/artifact/com.ververica/flink-connector-mysql-cdc/2.3.0) and [Amoro](https://mvnrepository.com/artifact/com.netease.arctic/arctic-flink-runtime-1.14/0.4.1) dependencies to your Maven project's pom.xml file.
```
Comment (huyuanfeng2018) [inline docs/engines/flink/flink-cdc-ingestion.md:344]: `https://mvnrepository.com/artifact/com.netease.arctic/arctic-flink-runtime-1.14/0.4.1`
Is this link correct?

**Review by huyuanfeng2018 — COMMENTED** [review]
I left some comments, please take a look.

**Review by huyuanfeng2018 — APPROVED** [review]
LGTM.

**Pull request merged**